src/cromshell/metadata/command.py

```diff
@@ -0,0 +1,124 @@
+import logging
+
+import click
+import requests
+
+from cromshell.utilities import cromshellconfig, http_utils, io_utils
+
+LOGGER = logging.getLogger(__name__)
+
+
+@click.command(name="metadata")
+@click.argument("workflow_id")
+@click.option(
+    "-des",
+    "--dont-expand-subworkflows",
+    is_flag=True,
+    default=True,
+    help="Do not expand subworkflow info in metadata",
+)
+@click.pass_obj
+def main(config, workflow_id: str, dont_expand_subworkflows: bool):
+    """Get the full metadata of a workflow."""
+
+    LOGGER.info("metadata")
+
+    check_cromwell_server(config=config, workflow_id=workflow_id)
+
+    obtain_and_print_metadata(
+        config=config,
+        metadata_param=config.METADATA_PARAMETERS,
+        exclude_keys=True,
+        expand_subworkflows=dont_expand_subworkflows,
+    )
+
+    return 0
+
+
+def check_cromwell_server(config, workflow_id):
+    """Checks for an associated cromwell server for the workflow_id
+    and checks connection with the cromwell server"""
+
+    # Overrides the default cromwell url set in the cromshell config file or
+    # command line argument if the workflow id is found in the submission file.
+    cromshellconfig.resolve_cromwell_config_server_address(workflow_id=workflow_id)
+
+    config.cromwell_api_workflow_id = f"{config.cromwell_api}/{workflow_id}"
+
+    # Check if Cromwell Server Backend works
+    http_utils.assert_can_communicate_with_server(config)
+
+
+def format_metadata_params(
+    list_of_keys: list, exclude_keys: bool, expand_subworkflow: bool
+) -> dict:
+    """This functions organises a list of cromwell metadata keys and flags into a
+    dictionary that can passed to requests library"""
+
+    if not list_of_keys:
+        LOGGER.error("Function format_metadata_params was given an empty list.")
+        raise ValueError("Function format_metadata_params was given an empty list.")
+    elif "" in list_of_keys:
+        LOGGER.error(
+            "Function format_metadata_params was given a list with empty element."
+        )
+        raise ValueError(
+            "Function format_metadata_params was given a list with empty element."
+        )
+    else:
+        # Determines whether the list of keys will be used to exclude or
+        # include fields in the metadata.
+        key_action = "excludeKey" if exclude_keys else "includeKey"
+
+        final_key = {key_action: list_of_keys}
+
+        if expand_subworkflow:
+            final_key["expandSubWorkflows"] = "true"
+
+        return final_key
+
+
+def get_workflow_metadata(
+    meta_params: dict,
+    api_workflow_id: str,
+    timeout: int,
+    verify_certs: bool,
+) -> str:
+    """Use requests to get the metadata or sub-metadata of
+    a workflow from the cromwell server."""
+
+    requests_out = requests.get(
+        f"{api_workflow_id}/metadata",
+        params=meta_params,
+        timeout=timeout,
+        verify=verify_certs,
```

Comment:
I'm always a bit worried if this intended-to-be temporary solution will be carried on indefinitely.
Can you add a todo to remove this, after certain dates, if this is not already a to-do item?

Reply:
also possible to hide the option. https://stackoverflow.com/questions/34334392/python-click-make-some-options-hidden

Reply:
nice, I guess we can just to make `hidden=true`  as the default